R/integration.R

```diff
@@ -139,3 +139,15 @@ runLiger = function(sobj, batch, hvg, k=20, res=0.4, small.clust.thresh=20) {
 
     return(sobj)
 }
+
+runFastMNN = function(sobj, batch) {
+	require(batchelor)
+
+	expr <- sobj@assays$RNA@data
+
+	sce <- fastMNN(expr, batch = sobj@meta.data[[batch]])
+
+	sobj@reductions['X_emb'] <- reducedDim(sce, "corrected")
```

Comment:
if that's an output of the method, then yes!

Reply:
There is a reconstructed expression matrix but the documentation suggests not to use it for further analysis (emphasis mine):

> A reconstructed matrix in the assays slot, containing the low-rank reconstruction of the expression matrix. **This can be interpreted as per-gene corrected log-expression values (after cosine normalization, if cos.norm=TRUE) but should not be used for quantitative analyses.**

I think for the other methods we didn't include outputs that weren't recommended? I don't really mind as long as it's consistent and clear in the paper.

Reply:
Quantitative analyses sounds like DE testing to me. I would not recommend any corrected expression data for DE testing. We would just be testing it for relative placement of cells in an embedding, and for cell cycle effect preservation and HVG conservation. What do you make of that interpretation @lazappi ?

Reply:
I just assumed it meant anything but you could be right. I could ask Aaron what he thinks so we are sure?

Reply:
That would be super helpful! We'll add it for now, and would remove if Aaron says no.